lib/moku/reference_repo.rb

```diff
@@ -22,6 +23,7 @@ def initialize(dir, runner = Moku.git_runner)
     # @return [Lazy::Directory]
     def resolve(ref)
       wd = checkout(ref, dir_for(ref))
+      cleanup!
```

Comment:
I think this can result in the active release being pruned if there are successive failed deployments. I think the best behavior would be like this with the exclusion of whichever is active -- so if we're over the cap and the oldest is active, we'd prune the second oldest.

Alternatively, we could only prune on success and offer a manual cleanup for those situations where a viable release is not practical for a while (after successive failed deployments). I guess that would be equivalent to redeploying whatever version is active, redundant as that may be, so a separate command wouldn't be strictly necessary.

I think the main scenario we're trying to avoid is lots of viable releases hanging around. Either implementation would get us there.

Reply:
Nice catch. I think pruning the active release from those marked for deletion would be sufficient.

Reply:
Oh, right! No problem, then.